### PR TITLE
Release: merge development into main

### PR DESCRIPTION
**Pull request by dimileeh**
Automated AWF release PR syncing `development` into `main`.

Opened by the `sync_release_pr` task kind and monitored with release/manual behavior (auto-merge disabled). Merging into the release target stays human-gated.

### Scope note — the `.awf/workspace.yml` dogfood override is included

This release PR **does** carry the `.awf/workspace.yml` change that restores AWF's dogfood
auto-merge default through `monitor.auto_merge.by_base_branch.development: true` (base `main`
stays opt-out, which is why this PR itself is human-gated above). The override is
operator-authorized under the AGENTS.md "AWF Dogfood Rules" requirement that AWF work targeting
a non-`main` branch auto-merges by default, and it is intentionally part of this release rather
than deferred. Issue [#794](https://github.com/dimileeh/agent-workspace-fabric/issues/794),
which previously tracked the override as follow-up work, is now CLOSED as done by this PR. As a
protected quality-gate file edit it stays subject to AWF's operator-approval pause on push.

---

> [!NOTE]
> **High Risk**
> Changes default merge behavior (breaking), centralizes merge authorization at provision time, and adds gates for retargeted PRs and adopted PR profile trust—incorrect resolution could merge when operators expect manual control.
> 
> **Overview**
> **Breaking:** `auto_merge` is now **opt-in and defaults to off** for workspace create, PR adoption, CLI, MCP, and new DB rows. `task_kind` (including `sync_release_pr`) no longer forces merge behavior; the persisted `workspace.auto_merge` column—resolved once at provision from tri-state intent (`--auto-merge` / omit), `monitor.auto_merge` in `workspace.yml` (`default` + `by_base_branch`), then `false`—is the sole monitor switch (feature squash-merge vs manual “ready to merge”).
> 
> Create/adopt persist `task_policy.auto_merge_intent`; APIs report `auto_merge` as **`null` until resolved** so clients do not see a provisional manual policy. Legacy rows without the intent key keep their stored column; an Alembic migration flips the column **server default** to `false`. Repo dogfood restores `monitor.auto_merge.by_base_branch.development: true` while `main` stays off.
> 
> **Merge safety:** PRs **retargeted after provisioning** (`base_ref` ≠ `branch_base`) trigger `NotifyHuman` instead of merge/sync under stale base policy. Adopted feature PRs whose profile comes from the **PR head checkout** cannot enable auto-merge from profile alone—only explicit operator intent. `sync_release_pr` workspaces that resolve `auto_merge=True` use the feature monitor but **do not delete** the long-lived source branch on merge; reused release PR bodies reconcile only AWF’s marker-delimited merge-policy section (with malformed-marker fail-closed).
> 
> Smaller refactors: workspace I/O schemas and Bitbucket HTTP transport split for line limits; forge clients populate `base_ref` on PR status.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 69e4ef2cc0ca38acb955ab320691724ee41427c0. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Opt-in, default-off auto-merge with global + per–base-branch configuration and per-workspace intent (tri-state until resolved).
  * PR monitor can now manage merge-policy text while preserving human-authored PR content.
  * Added control over whether successful merges delete the source branch.
* **Breaking Changes**
  * Global `auto_merge` now defaults to `false`; merge behavior is driven by resolved workspace settings (not task kind).
  * API/OpenAPI `auto_merge` now supports omitted/unresolved intent (`null`).
* **Bug Fixes**
  * Retargeted PRs pause for human review when the base branch drifts.
* **Documentation**
  * Updated changelog, README, CLI reference, concepts, and adoption/creation docs to match the new precedence and defaults.

### CODE REVIEW

**Comment by dimileeh** [comment]
⚠️ PR #792 needs human attention at commit `e6bc13768e`.

AWF did not auto-merge because a protected quality-gate file was changed and AWF paused the workspace for an operator decision: agent changed protected quality-gate file(s) outside declared owned_paths:
- .awf/workspace.yml :: .awf/workspace.yml :: line unknown :: protected quality-gate file changed outside declared owned_paths
AWF only allows narrowly classified safe protected-file edits without explicit ownership. Add meaningful tests or declare explicit ownership of the policy file when an operator-approved policy change is required.
Note: editing AWF validation config in .awf/workspace.yml has no effect mid-run — `requested_tier` is set at dispatch (validation.requested_tier) and `final_gate` is fixed at provision time. Do not edit the file to raise the AWF validation level; a validation command the repo delegates to CI is gated by CI, not AWF..

After the blocker is cleared or a new commit lands, AWF will re-verify the PR before taking any merge action.

File: src/awf/runtime/release_pr_sync.py [hunk 1]
```diff
@@ -402,10 +427,123 @@ def release_pr_title(*, source_branch: str, target_branch: str) -> str:
...
+    reconciliation would append the fresh section beside the stale paragraph,
+    yielding a PR that simultaneously claims manual and automatic merge. Delete the
+    exact generated paragraph text (both variants) while preserving surrounding
+    human-authored content.
+    """
+
+    for auto_merge in (False, True):
+        legacy = _release_merge_policy_text(auto_merge=auto_merge)
+        body = body.replace(legacy, "")
+    return body
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/runtime/release_pr_sync.py]: DEFER: Valid edge-case follow-up, but this is a clean development→main release PR and must not introduce new code changes. Tracking as a separate feature PR to `development`: https://github.com/dimileeh/agent-workspace-fabric/issues/793

AWF deferred the review thread on `src/awf/runtime/release_pr_sync.py` and filed https://github.com/dimileeh/agent-workspace-fabric/issues/793 to track the follow-up. Resolving this thread; the deferred work lives in that issue.

File: src/awf/cli/workspace_commands.py [hunk 2]
```diff
@@ -142,10 +142,13 @@ def workspace_create(
         "--with-db",
         help="Deprecated v1 shortcut; selects the aira profile when set.",
     ),
-    auto_merge: bool = typer.Option(
-        True,
+    auto_merge: bool | None = typer.Option(
+        None,
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/cli/workspace_commands.py:146]: DEFER: Dogfood `by_base_branch.development: true` belongs in a separate feature PR to `development`, not in this release promotion. Editing `.awf/workspace.yml` on a development→main release PR is explicitly out of scope (protected policy file; mid-run edits also do not change the in-flight validation tier). Tracking: https://github.com/dimileeh/agent-workspace-fabric/issues/794

AWF deferred the review thread on `src/awf/cli/workspace_commands.py` / `.awf/workspace.yml` and filed https://github.com/dimileeh/agent-workspace-fabric/issues/794 to track the follow-up. Resolving this thread; the deferred work lives in that issue.

**Comment by dimileeh** [comment]
✅ PR #792 is ready to merge at commit `e6bc13768e`.

All 5 AWF gates are green:
1. Inline comments resolved.
2. Outside-diff comments addressed.
3. CI checks all SUCCESS or SKIPPED.
4. Mergeable.
5. Base merged into head.

AWF will not merge this PR automatically — human action required. If new commits land here, AWF will re-verify all 5 gates and re-post this message on the new head SHA.

**Comment by dimileeh** [comment]
✅ PR #792 is ready to merge at commit `e6bc13768e`.

All 5 AWF gates are green:
1. Inline comments resolved.
2. Outside-diff comments addressed.
3. CI checks all SUCCESS or SKIPPED.
4. Mergeable.
5. Base merged into head.

AWF will not merge this PR automatically — human action required. If new commits land here, AWF will re-verify all 5 gates and re-post this message on the new head SHA.

File: src/awf/runtime/release_pr_sync.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/runtime/release_pr_sync.py]: FIXED in fe793143e (reversing the earlier DEFER at operator request).

`_strip_legacy_policy_paragraph` now anchors removal to whole lines
(`re.sub(rf"^{re.escape(legacy)}[ \t]*$", "", body, flags=re.MULTILINE)`), so only a
standalone AWF-authored paragraph is stripped. A legacy paragraph a human quoted
inside release notes is preserved: blockquoted (`> Opened by the ...`), embedded
mid-line with commentary around it, or inside a list item (where deleting the text
would also leave a dangling bullet).

Regression tests added in `tests/unit/runtime/test_release_pr_sync.py`:
- `test_preserves_legacy_paragraph_quoted_inside_human_content` — blockquoted and
  mid-line-embedded copies both survive, and the managed section is still spliced once.
- `test_strips_legacy_paragraph_only_when_it_owns_its_line` — the list-item copy stays,
  the standalone stale paragraph goes.

Focused local checks: `pytest tests/unit/runtime/test_release_pr_sync.py` (43 passed),
`tests/unit/control/test_executor_error_paths_parts/test_executor_error_paths_part_007.py`
(23 passed), plus ruff check/format and mypy on the touched files. Broad validation,
coverage, and merge gating are owned by AWF/CI after the agent phase.

File: src/awf/cli/workspace_commands.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/cli/workspace_commands.py:146]: Acknowledged, and the concern is valid — but this PR is not the place to land it.

The operator is landing the dogfood override
(`monitor.auto_merge.by_base_branch.development: true` in `.awf/workspace.yml`)
separately, so this workspace made no edit to `.awf/workspace.yml` or any other
protected quality-gate file. Tracking remains
https://github.com/dimileeh/agent-workspace-fabric/issues/794.

No code change here; keeping this thread resolved.

**Comment by dimileeh** [comment]
### Status update — Greptile P2 addressed (defer reversed)

Per operator instruction, the deferred Greptile P2 on `src/awf/runtime/release_pr_sync.py`
is now fixed on this branch rather than tracked as follow-up work:

- `_strip_legacy_policy_paragraph` removes a pre-marker AWF merge-policy paragraph only
  when it is **standalone** (alone on its own line). A legacy paragraph a human quoted
  inside release notes — blockquoted, embedded mid-line, or in a list item — is preserved.
- Two regression tests cover the quoted/embedded and list-item-vs-standalone cases.

No `.awf/workspace.yml` or other protected quality-gate file was touched: the operator is
landing the `monitor.auto_merge.by_base_branch.development` dogfood override separately
(https://github.com/dimileeh/agent-workspace-fabric/issues/794).

Focused local checks passed (`tests/unit/runtime/test_release_pr_sync.py` 43 passed,
`test_executor_error_paths_part_007.py` 23 passed, ruff + mypy on the touched files).
Full validation and coverage gating are owned by AWF/CI.

**This is a release PR into `main`, so the merge stays human-gated: ready for a human to
merge once CI is green on the new commit.**

**Comment by dimileeh** [comment]
AWF deferred the review thread on `src/awf/service/workspaces_create.py` and filed https://github.com/dimileeh/agent-workspace-fabric/issues/795 to track the follow-up. Resolving this thread; the deferred work lives in that issue.

File: src/awf/common/github_client.py [hunk 3]
```diff
@@ -832,6 +832,59 @@ async def post_comment(self, *, repo: RepoRef, pr_number: int, body: str) -> Non
...
+                repo.slug(),
+                "--json",
+                "body",
+                "--jq",
+                ".body",
+            ],
+            operation="gh pr view body",
+            retry_policy=RetryPolicy.READ,
+        )
+        return result.stdout.rstrip("\n")
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/common/github_client.py]: FIXED in 5a7d4feff.

Confirmed valid: `gh pr view --json body --jq .body` renders a JSON `null` body as the
literal text `null`, and `fetch_pull_request_body` is read only by
`find_or_create_release_pr`, so for a description-less reused release PR that sentinel
became the "existing body" and was written back as a description reading `null`.

Fixed at the source by filtering with `.body // ""`, mirroring the `// empty` fallback the
post-merge SHA read in the same client already uses. Filtering in jq rather than
normalizing the rendered stdout also preserves a description whose literal text *is*
`null` — once `gh` has rendered it, the sentinel and that body are indistinguishable.

Regression tests in `tests/unit/common/test_github_client_parts/test_github_client_part_007.py`
(the method previously had no direct unit test):
- `test_fetch_pull_request_body_argv_maps_missing_description_to_empty` — pins the
  `.body // ""` filter and the empty return (`FakeCommandRunner` cannot evaluate jq, so the
  argv assertion is what locks the fix).
- `test_fetch_pull_request_body_preserves_internal_formatting` — only gh's single trailing
  newline is stripped; internal blank lines survive.

Focused local checks: `pytest tests/unit/common/test_github_client_parts/test_github_client_part_007.py
tests/unit/runtime/test_release_pr_sync.py` (82 passed), plus ruff check/format and mypy.
Broad validation, coverage, and merge gating are owned by AWF/CI after the agent phase.

File: tests/unit/common/test_auto_merge.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+@pytest.mark.parametrize("intent", [True, False])
+def test_seed_auto_merge_keeps_explicit_intent(intent: bool) -> None:
+    assert seed_auto_merge(intent) is intent
+
+
+def test_seed_auto_merge_unset_intent_uses_opt_in_default() -> None:
+    # create/adopt seed the column before the profile resolves, so an unset
+    # intent must persist the conservative default rather than guess.
+    assert seed_auto_merge(None) is DEFAULT_AUTO_MERGE
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply tests/unit/common/test_auto_merge.py:75]: FIXED in 9f35a1898.

Valid: the module carried no marker, so marker-based selection skipped every case in it.
Added module-scope `pytestmark = pytest.mark.unit`, matching the convention in the other
`tests/unit/common` modules.

Verified with `pytest tests/unit/common/test_auto_merge.py -q -m unit` — 13 selected and
passed (previously 0 selected under `-m unit`).

File: src/awf/runtime/release_pr_sync.py [hunk 5]
```diff
@@ -324,6 +331,25 @@ async def find_or_create_release_pr(
...
+        # body already matches so a no-op sync makes no write. Release-PR sync is
+        # GitHub-only (``ensure_release_sync_forge_supported`` above), so ``gh`` is a
+        # ``GitHubClient`` whose read/edit map to reason-coded failures.
+        assert isinstance(gh, GitHubClient)
+        existing_body = await gh.fetch_pull_request_body(repo=repo, pr_number=pr_number)
+        reconciled_body = reconcile_release_pr_body(
+            existing_body=existing_body, generated_body=body
+        )
+        if reconciled_body != existing_body:
+            await gh.update_pull_request_body(repo=repo, pr_number=pr_number, body=reconciled_body)
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/runtime/release_pr_sync.py:352]: DEFER — valid, tracked as https://github.com/dimileeh/agent-workspace-fabric/issues/796

Confirmed: `_create_release_pr_with_redundancy` returns `created=False` when it adopts a PR
after a duplicate/transient `gh pr create` failure, and the fetch/reconcile/edit block lives
only in the `existing_number is not None` branch, so a raced adoption keeps its stale policy
paragraph while the attached monitor enforces the resolved `auto_merge`.

Not fixed here: this PR is a `development` → `main` release promotion and the operator scoped
this fix cycle to two specific review findings. The fix changes release-sync behavior on the
raced path (an extra read plus a conditional write) and wants regression coverage for both
adoption routes, so it goes in its own PR to `development`.

File: src/awf/runtime/release_pr_sync.py [hunk 6]
```diff
@@ -402,10 +428,179 @@ def release_pr_title(*, source_branch: str, target_branch: str) -> str:
...
+    section = _extract_managed_policy_section(generated_body)
+    if section is None:
+        # Defensive: a body without managed markers can't be reconciled section-wise.
+        # Never seen in practice (``release_pr_body`` always emits them); fall back to
+        # the generated body rather than silently dropping the policy statement.
+        return generated_body
+    start = existing_body.find(_MANAGED_POLICY_START)
+    end = existing_body.find(_MANAGED_POLICY_END)
+    start_count = existing_body.count(_MANAGED_POLICY_START)
+    end_count = existing_body.count(_MANAGED_POLICY_END)
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/runtime/release_pr_sync.py]: DEFER — valid, tracked as https://github.com/dimileeh/agent-workspace-fabric/issues/797

Confirmed: `_extract_managed_policy_section` and the marker counting in
`reconcile_release_pr_body` use unrestricted `find`/`count`, so a marker pair a human quoted
inside a fenced block is treated as the live managed section — the sample text is rewritten
and no real policy section is appended. `_strip_legacy_policy_paragraph` is already
container-aware (`_FENCE_OPEN_RE`, `_fence_closes`, `_advance_html_comment_state`), so the
right fix is to share that scan.

Not fixed here: this PR is a `development` → `main` release promotion and the operator scoped
this fix cycle to two specific review findings. Making the marker lookup container-aware also
changes the splice offsets and the fail-closed `RELEASE_SYNC_POLICY_MARKERS_MALFORMED` guard
(quoted pairs must not be counted), which is more than a release promotion should carry.

File: src/awf/service/pr_monitor_adoption.py [hunk 7]
```diff
@@ -537,6 +549,31 @@ async def _response(
...
+        # *preserves* the persisted ``auto_merge`` column rather than re-resolving
+        # it, so that column is already the authoritative policy even while the row
+        # is still ``requested``/``provisioning``. Reporting such a row as
+        # unresolved would hide a grandfathered ``True`` that will auto-merge, so
+        # treat a missing intent key as a legacy-resolved policy.
+        auto_merge_intent = auto_merge_intent_from_policy(workspace.task_policy)
+        legacy_policy = not task_policy_has_auto_merge_intent(workspace.task_policy)
+        auto_merge_resolved = (
+            workspace.status not in _AUTO_MERGE_UNRESOLVED_STATUSES or legacy_policy
+        )
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/service/pr_monitor_adoption.py]: DEFER — valid, tracked as https://github.com/dimileeh/agent-workspace-fabric/issues/798

Agreed that a workspace going straight from `requested`/`provisioning` to `failed`/`cancelled`
never reaches the profile resolver, so inferring "resolved" from a terminal status can
advertise a manual policy that a retry would resolve to `true`.

Not fixed here: this PR is a `development` → `main` release promotion and the operator scoped
this fix cycle to two specific review findings. Deriving resolution from the recorded
resolution event instead of the status vocabulary touches the adoption response contract, so
it goes in its own PR to `development`.

File: .awf/workspace.yml [hunk 8]
```diff
@@ -35,6 +35,12 @@ awf:
     non_check_reviewer_logins:
       - greptile-apps
       - chatgpt-codex-connector
+    # AGENTS.md "AWF Dogfood Rules": AWF work targeting a non-`main` branch must
+    # auto-merge by default. auto_merge now defaults off everywhere, so restore
+    # the required dogfood default via a per-base-branch override (main stays off).
+    auto_merge:
+      by_base_branch:
+        development: true
```
[the remark this answers is not in the document]
Comment (dimileeh) [reply .awf/workspace.yml:43]: Operator-owned — intentionally left as-is on this PR.

The scope note you are quoting was accurate when it was written: an earlier cycle deferred
the dogfood override to https://github.com/dimileeh/agent-workspace-fabric/issues/794. The
operator has since decided this file's fate directly and instructed this fix cycle to leave
the `.awf/workspace.yml` change in place, so no agent edit is made here either way. The
change is a protected quality-gate file edit, so AWF pauses the PR for an explicit operator
decision on push rather than merging it silently.

Worth recording for the human reviewing that decision: the override only restores the
non-`main` dogfood default that AGENTS.md requires (`main` stays opt-out), and editing this
file mid-run does not change the in-flight validation tier — `requested_tier` is fixed at
dispatch and `final_gate` at provision time.

File: src/awf/runtime/release_pr_sync.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dimileeh) [reply src/awf/runtime/release_pr_sync.py]: Addendum — the container-aware follow-up is already on the branch (commit `4be7b122e`,
newer than `ed2ad0bd0`, the commit the 4/5 summary reviewed), and I re-verified it this
cycle.

`_strip_legacy_policy_paragraph` now walks the body line by line and tracks two containers,
so a standalone match at column zero is only stripped at top level:
- fenced code blocks — `_FENCE_OPEN_RE` (backtick or tilde, up to three spaces of indent)
  with `_fence_closes` requiring a same-character fence at least as long;
- HTML comments — `_advance_html_comment_state`, which handles openers/closers appearing
  mid-line and multiple times per line.

Regression tests in `tests/unit/runtime/test_release_pr_sync.py`:
- `test_preserves_legacy_paragraph_inside_fenced_code_block` — a verbatim quote inside
  ```` ```text ```` survives while a standalone copy outside the fence is still stripped;
- `test_preserves_legacy_paragraph_inside_html_comment` — text parked in `<!-- ... -->`
  survives;
- `test_strips_legacy_paragraph_after_a_closed_fence` — an early code block does not shield
  every later stale paragraph.

Re-verified beyond those tests, by driving the helper directly over the container edge cases
a human body can realistically produce; the legacy paragraph is preserved for a 3-space
indented fence, a tilde fence with an info string, an over-long closing fence, an unclosed
fence, and a fence nested inside an HTML comment, and correctly stripped after a single-line
comment, after a comment that closes mid-line, after a fence opened inside a comment, and
after a `<!--` that appeared inside a fence (where no comment is actually open). Both
container-tracking mistakes are fail-safe in the conservative direction: an ambiguous
container stays open, so cleanup is skipped rather than human text deleted.

One related finding from this cycle is *not* fixed here and is tracked instead: the managed
*marker* lookup is still container-unaware, so markers quoted inside a fence are mistaken
for the live section — https://github.com/dimileeh/agent-workspace-fabric/issues/797.

Focused local checks: `pytest tests/unit/runtime/test_release_pr_sync.py` (43 passed), plus
ruff check/format and mypy. Broad validation, coverage, and merge gating are owned by AWF/CI.

**Comment by dimileeh** [comment]
### Status update — operator hint addressed

**1. `_strip_legacy_policy_paragraph` container awareness (Greptile 4/5) — already on the branch, re-verified.**
The fix landed in `4be7b122e`, which is newer than `ed2ad0bd0` (the commit the 4/5 summary
reviewed), so the summary predates it. The helper now tracks fenced code blocks and HTML
comments line by line and strips a standalone match only at top level; regression tests
cover a fenced quote, an HTML-comment quote, and stripping resuming after a closed fence.
I re-drove the helper over the realistic container edge cases (indented and tilde fences,
over-long and missing closing fences, comments closing mid-line, a fence inside a comment
and a `<!--` inside a fence) and every case behaves correctly, with ambiguity failing safe
toward *not* stripping. Details in the reply on that thread. No new code change needed.

**2. `gh --jq .body` returning the literal `null` (CodeRabbit Minor) — fixed in `5a7d4feff`.**
A description-less PR made `fetch_pull_request_body` return the string `null`, which flowed
into reconciliation and could be written back as a body reading "null". The filter is now
`.body // ""`, matching the `// empty` fallback already used by the post-merge SHA read in
the same client; filtering in jq also preserves a description whose literal text *is* `null`.
Two regression tests added — the method previously had no direct unit test.

`.awf/workspace.yml` was left untouched per the operator instruction; that decision stays
with the operator and AWF pauses the PR for it on push.

**Also this cycle:** `9f35a1898` adds the missing `pytestmark = pytest.mark.unit` to
`tests/unit/common/test_auto_merge.py` (CodeRabbit Minor; 13 cases were invisible to
marker-based selection). Three further review findings are valid but out of scope for a
`development` → `main` release promotion and are tracked instead — #796 (raced PR adoption
skips the body reconcile), #797 (managed marker lookup is container-unaware), and #798
(terminal pre-provision status reported as a resolved auto-merge policy). All review threads
are answered and resolved.

Focused local checks: `pytest tests/unit/common/test_github_client_parts/test_github_client_part_007.py
tests/unit/runtime/test_release_pr_sync.py` (82 passed), `pytest tests/unit/common/test_auto_merge.py -m unit`
(13 passed), ruff check/format and mypy clean. Broad validation, coverage, and merge gating
are owned by AWF/CI after the agent phase.

**This is a release PR into `main`, so the merge stays human-gated.**

**Comment by dimileeh** [comment]
AWF deferred the review thread on `src/awf/common/auto_merge.py` and filed https://github.com/dimileeh/agent-workspace-fabric/issues/799 to track the follow-up. Resolving this thread; the deferred work lives in that issue.

**Comment by dimileeh** [comment]
✅ PR #792 is ready to merge at commit `93a777d128`.

All 5 AWF gates are green:
1. Inline comments resolved.
2. Outside-diff comments addressed.
3. CI checks all SUCCESS or SKIPPED.
4. Mergeable.
5. Base merged into head.

AWF will not merge this PR automatically — human action required. If new commits land here, AWF will re-verify all 5 gates and re-post this message on the new head SHA.

File: .awf/workspace.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (dimileeh) [reply .awf/workspace.yml:43]: FALSE POSITIVE relative to the updated scope — the override is operator-authorized and
intentionally included in this release.

Correcting my earlier reply on this thread, which still framed the deferral as merely
superseded: the operator has now decided this file's fate explicitly. The
`monitor.auto_merge.by_base_branch.development: true` override in `.awf/workspace.yml` is
authorized under AGENTS.md "AWF Dogfood Rules" (L108) — AWF work targeting a non-`main` branch
must auto-merge by default — and ships as part of this release. It is not deferred.

Issue [#794](https://github.com/dimileeh/agent-workspace-fabric/issues/794), the follow-up this
finding cites, is now **CLOSED as done**, so the scope premise behind the suggestion no longer
holds. I have also updated the PR body: it now carries a "Scope note" stating the override is
included and operator-authorized, and it no longer implies `.awf/workspace.yml` is unmodified or
that the override waits on #794.

Unchanged from before, and worth keeping in front of the human who approves this: base `main`
stays opt-out, so this release PR itself remains human-gated; the edit is a protected
quality-gate file change, so AWF pauses the PR for an explicit operator decision on push rather
than merging it silently; and editing this file mid-run does not shift the in-flight validation
tier, since `requested_tier` is fixed at dispatch and `final_gate` at provision time.

No code change on this thread — resolving it as a false positive.

**Comment by dimileeh** [comment]
### Status update — operator hint addressed (workspace.yml decision recorded)

**1. `.awf/workspace.yml` override — KEPT, unchanged.**
Per the operator decision, the `monitor.auto_merge.by_base_branch.development: true` override
stays on this branch. No edit was made to the file in this cycle.

**2. CodeRabbit finding on `.awf/workspace.yml` — replied and resolved as FALSE POSITIVE.**
[Reply](https://github.com/dimileeh/agent-workspace-fabric/pull/792#discussion_r3679746730)
records that the override is operator-authorized under AGENTS.md "AWF Dogfood Rules" and
intentionally included in this release, and that
[#794](https://github.com/dimileeh/agent-workspace-fabric/issues/794) is now CLOSED as done, so
the deferral premise behind the suggestion no longer holds.

**3. PR body updated.**
The description now carries a "Scope note" stating the dogfood override is included and
operator-authorized (base `main` stays opt-out, so this release PR itself remains human-gated).
It no longer implies `.awf/workspace.yml` is unmodified or that the override is deferred to #794.

**4. Prior-directive code fixes re-verified in place — no new code needed.**
- Greptile fenced-code edge on `src/awf/runtime/release_pr_sync.py`: `_strip_legacy_policy_paragraph`
  and `_managed_marker_offsets` both track fenced code blocks (backtick and tilde, up to
  three-space indent) and HTML comments line by line, stripping or counting only standalone
  top-level matches. Regression tests cover a legacy paragraph quoted inside a fence, inside an
  HTML comment, and inside human prose, plus marker pairs quoted in a fence, a blockquote, and a
  multi-line HTML comment.
- `src/awf/common/github_client.py` null body: `fetch_pull_request_body` filters with
  `.body // ""`, so a description-less reused release PR can never round-trip the literal string
  `null` into its body.

Focused local checks: `pytest tests/unit/runtime/test_release_pr_sync.py
tests/unit/common/test_github_client_parts/test_github_client_part_007.py` — 90 passed. Broad
validation, coverage, and merge gating remain owned by AWF/GitHub CI after the agent phase.

**Comment by dimileeh** [comment]
✅ PR #792 is ready to merge at commit `69e4ef2cc0`.

All 5 AWF gates are green:
1. Inline comments resolved.
2. Outside-diff comments addressed.
3. CI checks all SUCCESS or SKIPPED.
4. Mergeable.
5. Base merged into head.

AWF will not merge this PR automatically — human action required. If new commits land here, AWF will re-verify all 5 gates and re-post this message on the new head SHA.
